src/Plutarch/Test/Export.hs

```diff
@@ -0,0 +1,728 @@
+ export.
+
+ @since 1.2.1
+-}
+settingScriptParams' ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (a :: Type).
+  -- | A \'local value\' for the arguments
+  ScriptParams dat red ->
+  -- | The collection of tests to run using the set arguments
+  WithExport dat red info a ->
+  WithExport dat red info a
+settingScriptParams' x = modifyingScriptParams (\_ _ -> x)
+
+-- Helpers
+
+runWithExport ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (param :: Type).
+  WithExport dat red info () ->
+  RawScriptExport ->
+  Linker param (ScriptExport info) ->
+  TestParams dat red param ->
+  Either LinkerError (Map Text PreparedScriptTests)
+runWithExport (WithExport comp) raw linker args =
+  case runLinker linker raw . view #linkerParams $ args of
+    Left err -> Left err
+    Right linked ->
+      let env =
+            WithExportEnv
+              (view (#scriptParams %% #datum) args)
+              (view (#scriptParams %% #redeemer) args)
+              (view (#scriptParams %% #context) args)
+              raw
+              (view #information linked)
+       in Right . coerce . snd . execRWS comp env $ ()
+
+-- Note from Koz: the manual lenses are due to a weird interaction between TH
+-- and record syntax. I wasn't able to establish _why_ it was being weird, but
+-- when written this way, it behaves.
+data WithExportEnv (dat :: Type) (red :: Type) (info :: Type)
+  = WithExportEnv dat red ScriptContext RawScriptExport info
+
+instance
+  (k ~ A_Lens, a ~ dat, b ~ dat) =>
+  LabelOptic "datum" k (WithExportEnv dat red info) (WithExportEnv dat red info) a b
+  where
+  labelOptic = lens (\(WithExportEnv x _ _ _ _) -> x) $ \(WithExportEnv _ r sc ra i) d' ->
+    WithExportEnv d' r sc ra i
+
+instance
+  (k ~ A_Lens, a ~ red, b ~ red) =>
+  LabelOptic "redeemer" k (WithExportEnv dat red info) (WithExportEnv dat red info) a b
+  where
+  labelOptic = lens (\(WithExportEnv _ x _ _ _) -> x) $ \(WithExportEnv d _ sc ra i) r' ->
+    WithExportEnv d r' sc ra i
+
+instance
+  (k ~ A_Lens, a ~ ScriptContext, b ~ ScriptContext) =>
+  LabelOptic "context" k (WithExportEnv dat red info) (WithExportEnv dat red info) a b
+  where
+  labelOptic = lens (\(WithExportEnv _ _ x _ _) -> x) $ \(WithExportEnv d r _ ra i) sc' ->
+    WithExportEnv d r sc' ra i
+
+instance
+  (k ~ A_Lens, a ~ RawScriptExport, b ~ RawScriptExport) =>
+  LabelOptic "raws" k (WithExportEnv dat red info) (WithExportEnv dat red info) a b
+  where
+  labelOptic = lens (\(WithExportEnv _ _ _ x _) -> x) $ \(WithExportEnv d r sc _ i) raws' ->
+    WithExportEnv d r sc raws' i
+
+instance
+  (k ~ A_Lens, a ~ info, b ~ info) =>
+  LabelOptic "info" k (WithExportEnv dat red info) (WithExportEnv dat red info) a b
+  where
+  labelOptic = lens (\(WithExportEnv _ _ _ _ x) -> x) $ \(WithExportEnv d r sc ra _) i' ->
+    WithExportEnv d r sc ra i'
+
+data Applied
+  = AppliedValidator Script Data Data Data
+  | AppliedMP Script Data Data
```

Comment:
Spell out MP, I think it's in standards for naming

Reply:
Actually, we have `modifySC` and it's friends. Hmmm. I like spelled out names, but it would be quite tedious if we enforce standards here. I'd say we just go with this.

Reply:
Yeah, the _style guide_ (as opposed to the standards) is a little more lax -- I think MP is fine here, but if anyone else disagreed I would defer to the expanded name.

Reply:
I would say that we should add a haddock comment spelling out the abbreviation to keep things clear.